_posts/2020-11-23-boost-your-productivity-with-tailwind-css.md

```diff
@@ -0,0 +1,184 @@
+---
+layout: post
+title: Boost your productivity with Tailwind CSS
+author: tuk
+excerpt: Tailwind CSS has proven its worth, and it's time you adopt it into your front-end toolkit. Let this post be your first step in doing so.
+tags:
+ - Software Development
+ - frontend
+ - CSS
+---
+
+If you've been following the front-end development scene lately, you may have heard a mention or two about [Tailwind CSS](https://tailwindcss.com/). The project has gained significant ground, and the trend seems to continue in its favor. In this blog post, I'll explain:
+
+* History of Tailwind CSS
+* What is Tailwind CSS
+* How to use Tailwind CSS
+
+I like to begin with history so let's jump into that. You may go ahead with examples if you feel like it.
+
+[![The State of CSS 2019](/img/boost-your-productivity-with-tailwind-css/state-of-css-scatterplot.png)](https://2019.stateofcss.com/technologies/#tools-scatterplot)
+
+## [History](https://adamwathan.me/tailwindcss-from-side-project-byproduct-to-multi-mullion-dollar-business/)
+
+Tailwind was born off work by Adam Wathan and Steve Schoger, who you might know from their other popular projects, such as [Refactoring UI](https://refactoringui.com/). The team behind the project released it on [GitHub](https://github.com/tailwindlabs/tailwindcss) on the 1st of November 2017.
+
+Tailwind has a curious background story because Wathan was publicly working on [KiteTail](https://kitetail.co/) on his YouTube channel before its initial release. The viewers of those videos encouraged Wathan and Schoger to open source their ["Less framework."](https://twitter.com/adamwathan/status/876256424694878208) Some of Wathan's friends and followers also found the utility-first aspect of it intriguing, namely Jonathan Reinink and Stefan Bauer, and offered to help. Less didn't quite bend in the way of what was required, so the project moved on to use PostCSS instead, as David Hemphill suggested. Afterward came v0.1.0 launch time.
+
+Since then, Tailwind has reached v2.0.1, and all but Bauer still work on Tailwind. It now includes plenty of essential features such as:
+
+* Purging of unused rules with PurgeCSS
+* Screenreader utilities
+* CSS transitions
+* CSS grid
+* Plugins
+* Animations
+* Dark mode
+
+## Examples
+
+Enough of the past. Let's get to the good stuff; Demos! We'll be using something called CodeSandbox, which is a fantastic tool, and I recommend it to you wholeheartedly. Unfortunately, you can not edit the embed below directly. If you wish to do that, push the "Open Sandbox" button and log in to CodeSandbox. The template I'm using uses Parcel for bundling, so it already includes PostCSS. To create the UI, the template features React v17.0.1 and Tailwind CSS v1.9.6. You could also use Angular, Svelte, Vue, or whatever for building components as well. If you want to quickly try out Tailwind CSS and don't want to hassle with JavaScript components, you could use [Tailwind Play](https://play.tailwindcss.com/).
```

Comment:
The link here is very good, almost offers interactive opportunity to play as you read - what I'm doing now.